x-pack/plugins/security_solution/server/lib/entity_analytics/risk_score/routes/entity_calculation.ts

```diff
@@ -0,0 +1,153 @@
+
+export const riskScoreEntityCalculationRoute = (
+  router: EntityAnalyticsRoutesDeps['router'],
+  getStartServices: EntityAnalyticsRoutesDeps['getStartServices'],
+  logger: Logger
+) => {
+  router.versioned
+    .post({
+      path: RISK_SCORE_ENTITY_CALCULATION_URL,
+      access: 'internal',
+      options: {
+        tags: ['access:securitySolution', `access:${APP_ID}-entity-analytics`],
+      },
+    })
+    .addVersion(
+      {
+        version: '1',
+        validate: {
+          request: { body: buildRouteValidation(riskScoreEntityCalculationRequestSchema) },
+        },
+      },
+      withRiskEnginePrivilegeCheck(getStartServices, async (context, request, response) => {
+        const securityContext = await context.securitySolution;
+
+        securityContext.getAuditLogger()?.log({
+          message: 'User triggered custom manual scoring',
+          event: {
+            action: RiskScoreAuditActions.RISK_ENGINE_ENTITY_MANUAL_SCORING,
+            category: AUDIT_CATEGORY.DATABASE,
+            type: AUDIT_TYPE.CHANGE,
+            outcome: AUDIT_OUTCOME.UNKNOWN,
+          },
+        });
+
+        const coreContext = await context.core;
+        const securityConfig = await securityContext.getConfig();
+        const siemResponse = buildSiemResponse(response);
+        const soClient = coreContext.savedObjects.client;
+
+        const riskScoreService = buildRiskScoreServiceForRequest(
+          securityContext,
+          coreContext,
+          logger
+        );
+
+        const { identifier_type: identifierType, identifier } = request.body;
+
+        try {
+          const entityAnalyticsConfig = await riskScoreService.getConfigurationWithDefaults(
+            securityConfig.entityAnalytics
+          );
+
+          if (entityAnalyticsConfig == null) {
+            return siemResponse.error({
+              statusCode: 405,
+              body: 'No Risk engine configuration found',
+            });
+          }
+
+          const {
+            dataViewId,
+            enabled,
+            range: configuredRange,
+            pageSize,
+            alertSampleSizePerShard,
+          } = entityAnalyticsConfig;
+
+          if (!enabled) {
+            return siemResponse.error({
+              statusCode: 405,
+              body: 'Risk engine is disabled',
+            });
+          }
+
+          const { index, runtimeMappings } = await getRiskInputsIndex({
+            dataViewId,
+            logger,
+            soClient,
+          });
+
+          const range = convertRangeToISO(configuredRange);
+
+          const afterKeys: AfterKeys = {};
+
+          const result: CalculateAndPersistScoresResponse =
+            await riskScoreService.calculateAndPersistScores({
+              pageSize,
+              identifierType,
+              index,
+              filter: { term: { [getFieldForIdentifier(identifierType)]: identifier } },
```

Comment:
I am not sure if this is right...
Should we merge the filter inside the risk engine configuration to the entity filter?

Reply:
I think we probably should include it, I could imagine customers wanting to exclude a certain type of alert for example? It seems safer to merge it the same way we do here: 
https://github.com/elastic/kibana/blob/abb8f6bb31f32a8f1e96536f752e395d99803bc2/x-pack/plugins/security_solution/server/lib/entity_analytics/risk_score/calculate_risk_scores.ts#L313

Reply:
If we do add it we should have an API test for it too.